### PR TITLE
[ext-doc] Consolidate "What are extensions?" into "Extensions 101" and "Development Basics"

### LINKED ISSUE #4322: What are Extensions? (Overview Refresh)
Rewrite the general extensions overview so that it can work better with Extensions 101, Development Basics, and Architecture Overview.

### PR DESCRIPTION
**Pull request by AmySteam**
Fixes #4322 

Extensions 101 and Development basics include most of the information included in [What are extensions?](https://developer.chrome.com/docs/extensions/mv3/overview/); therefore, this doc has become redundant. 

Changes proposed in this pull request:

- Consolidate any missing information into Extensions 101.
- Fix all the links that pointed to this doc. 
- Add a redirect from overview to architecture-overview.

Preview links
- https://deploy-preview-4533--developer-chrome-com.netlify.app/docs/extensions/mv3/getstarted/extensions-101/
- https://deploy-preview-4533--developer-chrome-com.netlify.app/docs/extensions/mv3/override/

### CODE REVIEW

File: site/en/docs/extensions/mv3/getstarted/extensions-101/index.md [hunk 1]
```diff
@@ -7,11 +7,19 @@ date: 2022-10-04
...
-browser. This page introduces Chrome extension development and provides links to step-by-step
-[beginner tutorials][section-tutorials]. 
+browser, providing things like:
+
+- Productivity tools.
+- Web page content enrichment.
+- Information aggregation.
+
+These are just a few examples of the many things that extensions can do. See the [Chrome Web
+Store][chrome-web-store] to see thousands of different examples of published extensions.
```
Comment (dotproto) [inline site/en/docs/extensions/mv3/getstarted/extensions-101/index.md:22]: "See" is currently used twice in quick succession.

```suggestion
These are just a few examples of the many things that extensions can do. Visit the [Chrome Web
Store][chrome-web-store] to see thousands of different examples of published extensions.
```

Alternatively…

```suggestion
These are just a few examples of the many things that extensions can do. See the [Chrome Web
Store][chrome-web-store] for thousands of different examples of published extensions.
```

**Review by dotproto — COMMENTED** [review]
The changes look good, but I'm a little hesitant about the impact this has on the site's TOC.

@AmySteam, are you thinking that this is part of a larger effort to get rid of the "Overview" section in the site nav? If not, I'm a worried that someone new to extension development might come to the site, open the "Overview" section, and be surprised/confused that there isn't something to explain the basics of what an extension is or why they'd want to build one.

That said, I don't want to block cleanup or organization improvements. Maybe we should chat a bit more about this in January?

**Comment by AmySteam** [comment]
> That said, I don't want to block cleanup or organization improvements. Maybe we should chat a bit more about this in January?

I agree. I have been working with @jpmedley on this, so maybe we can schedule a meeting to talk more about this effort.

**Comment by jpmedley** [comment]
Before we call a meeting, let me think about this a bit. I might have a way out.

**Comment by jpmedley** [comment]
@AmySteam Can you please resolve the merge conflicts. I don't want this to get too stale. (I can't see the conflicts. If they're too extensive, it might be better to open a new PR later.)

**Comment by AmySteam** [comment]
> @AmySteam Can you please resolve the merge conflicts. I don't want this to get too stale. (I can't see the conflicts. If they're too extensive, it might be better to open a new PR later.)

@jpmedley  I opened a [new PR](https://github.com/GoogleChrome/developer.chrome.com/pull/4782) to resolve these conflicts.

**Pull request closed without merging**